学生データのCRUD操作実装

AyanoCurious opened the pull request:
課題4回目

Commit:
Apache Commons Lang 追加

Commit:
Apache Commons Lang 追加

AyanoCurious commented:
名前と年齢を追加して登録できるようにしました。

KeitaUenishi commented on src/main/java/raisetech/studentManagement/StudentManagementApplication.java:
```diff
@@ -1,24 +1,49 @@
 
+  private Map<String, String> students = new HashMap<>();
+
+  {
+    students.put("Dale Cooper", "33");
+
+  }
```
ここのputって、どんな意図で書かれていますか？
不要ではないかと思いました！

AyanoCurious replied:
初期データも必要かと思って入れてしまいました。こちら削除します。

AyanoCurious commented:
![image](https://github.com/user-attachments/assets/e4216e1d-1df3-465d-8056-14d970c4d8b5)

AyanoCurious commented:
**gitbushの実行結果です**

Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Student not found.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Fafa&age=27'
Added: Fafa 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Fafa is 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Spencer&age=25'
Added: Spencer 25 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Watson&age=36'
Added: Watson 36 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Victroria&age=32'
Added: Victroria 32 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/allStudents'
{"timestamp":"2024-12-14T01:00:27.872+00:00","status":404,"error":"Not Found","p
ath":"/allStudents"}
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/allStudents'
{}
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Student not found.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Fafa&age=27'
Added: Fafa 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Fafa is 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Victroria&age=32'
Added: Victroria 32 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Spencer&age=25'
Added: Spencer 25 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Watson&age=36'
Added: Watson 36 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Watson'
Watson is 36 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/allStudents'
{"Fafa":"27","Victroria":"32","Watson":"36","Spencer":"25"}
Owner@Owner-PC MINGW64 ~
$

Commit:
Apache Commons Lang 追加

KeitaUenishi commented on src/main/java/raisetech/studentManagement/StudentManagementApplication.java:
```diff
@@ -1,24 +1,57 @@
   }
 }
+
+
+
+
+
+
+
+
+
```
50行目からの空行は消しておきましょう〜

KeitaUenishi commented on src/main/java/raisetech/studentManagement/StudentManagementApplication.java:
```diff
@@ -1,24 +1,57 @@
 
+  private Map<String, String> students = new HashMap<>();
+
+  {
+
+  }
+
```
この `{}` って意図的に残されています？
削除しておいた方がよいと思いました！

KeitaUenishi commented:
> gitbushの実行結果です

Git Bashですかね

コマンドの実行結果やソースコードについては以下のように、コードブロックを使うと見やすくなるのでぜひ使い方も調べてみてください 👍 

```
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Student not found.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Fafa&age=27'
Added: Fafa 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Fafa is 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Spencer&age=25'
Added: Spencer 25 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Watson&age=36'
Added: Watson 36 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Victroria&age=32'
Added: Victroria 32 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/allStudents'
{"timestamp":"2024-12-14T01:00:27.872+00:00","status":404,"error":"Not Found","p
ath":"/allStudents"}
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/allStudents'
{}
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Student not found.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Fafa&age=27'
Added: Fafa 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Fafa is 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Victroria&age=32'
Added: Victroria 32 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Spencer&age=25'
Added: Spencer 25 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Watson&age=36'
Added: Watson 36 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Watson'
Watson is 36 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/allStudents'
{"Fafa":"27","Victroria":"32","Watson":"36","Spencer":"25"}
Owner@Owner-PC MINGW64 ~
$
```

Commit:
Updated code with unused imports removed

AyanoCurious commented:
**Git Bashの実行結果**

```
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Student not found.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Fafa&age=27'
Added: Fafa 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Fafa is 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Spencer&age=25'
Added: Spencer 25 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Watson&age=36'
Added: Watson 36 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Victroria&age=32'
Added: Victroria 32 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/allStudents'
{"timestamp":"2024-12-14T01:00:27.872+00:00","status":404,"error":"Not Found","p
ath":"/allStudents"}
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/allStudents'
{}
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Student not found.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Fafa&age=27'
Added: Fafa 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Fafa'
Fafa is 27 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Victroria&age=32'
Added: Victroria 32 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Spencer&age=25'
Added: Spencer 25 years old.
Owner@Owner-PC MINGW64 ~
$ curl -X POST 'http://localhost:8080/studentInfo?name=Watson&age=36'
Added: Watson 36 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Watson'
Watson is 36 years old.
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/allStudents'
{"Fafa":"27","Victroria":"32","Watson":"36","Spencer":"25"}
Owner@Owner-PC MINGW64 ~
$
```
コードブロックを使用し、テキストと区別し見やすくしました。

Commit:
学生管理アプリケーションの機能を実装

AyanoCurious commented:
# Mysql
```
+-----------+------+
| name      | age  |
+-----------+------+
| Vivian    |   32 |
| Daisy     |   29 |
| Fafa      |   35 |
| Pino      |   41 |
| Catherine |   27 |
+-----------+------+
5 rows in set (0.02 sec)
```

KeitaUenishi commented:
> # Mysql
> ```
> +-----------+------+
> | name      | age  |
> +-----------+------+
> | Vivian    |   32 |
> | Daisy     |   29 |
> | Fafa      |   35 |
> | Pino      |   41 |
> | Catherine |   27 |
> +-----------+------+
> 5 rows in set (0.02 sec)
> ```

こちら、実際にアプリケーションからデータを取得するとどうなるかも合わせて確認させていただきたいです！
PR内にcurlの実行結果を添付、でOKです！

AyanoCurious commented:
**curl 実行結果**

```

Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/allStudents'
[{"name":"Vivian","age":32},{"name":"Daisy","age":29},{"name":"Fafa","age":35},{
"name":"Pino","age":41},{"name":"Catherine","age":27}]
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Daisy'
Daisy 29歳
Owner@Owner-PC MINGW64 ~
$ curl 'http://localhost:8080/studentInfo?name=Ayano'
Student not found.
Owner@Owner-PC MINGW64 ~
$

```

AyanoCurious commented:
セルフマージ完了しました！次回の課題も頑張ります！
